Issue installing scvi-tools

<!-- Describe the bug -->

Trying to install scvi-tools on python 3.10 using both conda and pip. The conda installation ends up not working with the solving environment repeatedly failing. The pip installation leads to an error where the building of the wheel for numba and llvmlite fails and causes a failure in installation.

<!-- To reproduce -->

```python
pip3 install scvi-tools
```

<!-- Put your Error output in this code block (if applicable, else delete the block): -->

```pytb
Building wheel for numba (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [803 lines of output]
      C:\Users\zacha\AppData\Local\Temp\pip-install-_2xfr3jd\numba_347914356f0e43d8896b5f2b75505b5f\setup.py:87: DeprecationWarning:

        `numpy.distutils` is deprecated since NumPy 1.23.0, as a result
        of the deprecation of `distutils` itself. It will be removed for
        Python >= 3.12. For older Python versions it will remain present.
        It is recommended to use `setuptools < 60.0` for those Python versions.
        For more details, see:
          https://numpy.org/devdocs/reference/distutils_status_migration.html


        import numpy.distutils.misc_util as np_misc
      TBB not found
      Using OpenMP from: True
      C:\Users\zacha\.conda\envs\CellAssign\lib\site-packages\setuptools\__init__.py:84: _DeprecatedInstaller: setuptools.installer and fetch_build_eggs are deprecated.
      !!

              ********************************************************************************
              Requirements should be satisfied by a PEP 517 installer.
              If you are using pip, you can try `pip install --use-pep517`.
              ********************************************************************************

      !!
        dist.fetch_build_eggs(dist.setup_requires)
      running bdist_wheel
      running build
      got version from file C:\Users\zacha\AppData\Local\Temp\pip-install-_2xfr3jd\numba_347914356f0e43d8896b5f2b75505b5f\numba/_version.py {'version': '0.51.2', 'full': '9d570961590c09a1eba748c9c37e91d1224fc9ad'}
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-310
      creating build\lib.win-amd64-cpython-310\numba
      running build_ext
      INFO: No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils
      building 'numba._dynfunc' extension
      creating build\temp.win-amd64-cpython-310
      creating build\temp.win-amd64-cpython-310\Release
      creating build\temp.win-amd64-cpython-310\Release\numba
      INFO: C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.36.32532\bin\HostX86\x64\cl.exe /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\Users\zacha\.conda\envs\C
ellAssign\include -IC:\Users\zacha\.conda\envs\CellAssign\Include -IC:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.36.32532\include -IC:\Program Files (x86)\
Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.36.32532\ATLMFC\include -IC:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Auxiliary\VS\include -IC:\Program Files (
x86)\Windows Kits\10\include\10.0.22000.0\ucrt -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\um -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\shared -IC:\
Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\winrt -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\cppwinrt -IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um /Tcnumba/_dynfuncmod.c /Fobuild\temp.win-amd64-cpython-310\Release\numba\_dynfuncmod.obj -g
      cl : Command line warning D9002 : ignoring unknown option '-g'
      INFO: C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.36.32532\bin\HostX86\x64\link.exe /nologo /INCREMENTAL:NO /LTCG /DLL /MANIFEST:EMBED,ID=2 /MANIFES
TUAC:NO /LIBPATH:C:\Users\zacha\.conda\envs\CellAssign\libs /LIBPATH:C:\Users\zacha\.conda\envs\CellAssign /LIBPATH:C:\Users\zacha\.conda\envs\CellAssign\PCbuild\amd64 /LIBPATH:C:\Program Fi
les (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.36.32532\ATLMFC\lib\x64 /LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.36.32532\l
ib\x64 /LIBPATH:C:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\lib\um\x64 /LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.22000.0\ucrt\x64 /LIBPATH:C:\Program Files (x86)\Windows 
Kits\10\\lib\10.0.22000.0\\um\x64 /EXPORT:PyInit__dynfunc build\temp.win-amd64-cpython-310\Release\numba\_dynfuncmod.obj /OUT:build\lib.win-amd64-cpython-310\numba\_dynfunc.cp310-win_amd64.pyd /IMPLIB:build\temp.win-amd64-cpython-310\Release\numba\_dynfunc.cp310-win_amd64.lib
      building 'numba._dispatcher' extension
      creating build\temp.win-amd64-cpython-310\Release\numba\core
      creating build\temp.win-amd64-cpython-310\Release\numba\core\typeconv
      INFO: C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.36.32532\bin\HostX86\x64\cl.exe /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\\Users\\zacha\\.conda\\en
vs\\CellAssign\\lib\\site-packages\\numpy\\core\\include -IC:\Users\zacha\.conda\envs\CellAssign\include -IC:\Users\zacha\.conda\envs\CellAssign\Include -IC:\Program Files (x86)\Microsoft Vi
sual Studio\2022\BuildTools\VC\Tools\MSVC\14.36.32532\include -IC:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.36.32532\ATLMFC\include -IC:\Program Files (x8
6)\Microsoft Visual Studio\2022\BuildTools\VC\Auxiliary\VS\include -IC:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.2
2000.0\\um -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\shared -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\winrt -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\cppwinrt -IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um /Tcnumba/_dispatcher.c /Fobuild\temp.win-amd64-cpython-310\Release\numba\_dispatcher.obj     
      _dispatcher.c
      numba/_dispatcher.c(26): error C2039: 'use_tracing': is not a member of '_ts'
      C:\Users\zacha\.conda\envs\CellAssign\include\cpython/pystate.h(60): note: see declaration of '_ts'
      numba/_dispatcher.c(28): error C2039: 'use_tracing': is not a member of '_ts'
      C:\Users\zacha\.conda\envs\CellAssign\include\cpython/pystate.h(60): note: see declaration of '_ts'
      numba/_dispatcher.c(199): warning C4244: '=': conversion from 'Py_ssize_t' to 'int', possible loss of data
      numba/_dispatcher.c(303): error C2039: 'use_tracing': is not a member of '_ts'
      C:\Users\zacha\.conda\envs\CellAssign\include\cpython/pystate.h(60): note: see declaration of '_ts'
      numba/_dispatcher.c(512): error C2039: 'use_tracing': is not a member of '_ts'
      C:\Users\zacha\.conda\envs\CellAssign\include\cpython/pystate.h(60): note: see declaration of '_ts'
      numba/_dispatcher.c(526): warning C4244: '=': conversion from 'Py_ssize_t' to 'int', possible loss of data
      error: Command "C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.36.32532\bin\HostX86\x64\cl.exe /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\\Users\\zacha\\
.conda\\envs\\CellAssign\\lib\\site-packages\\numpy\\core\\include -IC:\Users\zacha\.conda\envs\CellAssign\include -IC:\Users\zacha\.conda\envs\CellAssign\Include -IC:\Program Files (x86)\Mi
crosoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.36.32532\include -IC:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.36.32532\ATLMFC\include -IC:\Program
 Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Auxiliary\VS\include -IC:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt -IC:\Program Files (x86)\Windows Kits\10\\incl
ude\10.0.22000.0\\um -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\shared -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\winrt -IC:\Program Files (x86)\Win
dows Kits\10\\include\10.0.22000.0\\cppwinrt -IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um /Tcnumba/_dispatcher.c /Fobuild\temp.win-amd64-cpython-310\Release\numba\_dispatcher.obj" failed with exit status 2
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for numba
  Running setup.py clean for numba
  Building wheel for llvmlite (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [24 lines of output]
      running bdist_wheel
      C:\Users\zacha\.conda\envs\CellAssign\python.exe C:\Users\zacha\AppData\Local\Temp\pip-install-_2xfr3jd\llvmlite_a9473dabcc684c1a9cf12edd5b6d5d0a\ffi\build.py
      Trying generator 'Visual Studio 15 2017 Win64'
      Traceback (most recent call last):
        File "C:\Users\zacha\AppData\Local\Temp\pip-install-_2xfr3jd\llvmlite_a9473dabcc684c1a9cf12edd5b6d5d0a\ffi\build.py", line 191, in <module>
          main()
        File "C:\Users\zacha\AppData\Local\Temp\pip-install-_2xfr3jd\llvmlite_a9473dabcc684c1a9cf12edd5b6d5d0a\ffi\build.py", line 179, in main
          main_win32()
        File "C:\Users\zacha\AppData\Local\Temp\pip-install-_2xfr3jd\llvmlite_a9473dabcc684c1a9cf12edd5b6d5d0a\ffi\build.py", line 88, in main_win32
          generator = find_win32_generator()
        File "C:\Users\zacha\AppData\Local\Temp\pip-install-_2xfr3jd\llvmlite_a9473dabcc684c1a9cf12edd5b6d5d0a\ffi\build.py", line 76, in find_win32_generator
          try_cmake(cmake_dir, build_dir, generator)
        File "C:\Users\zacha\AppData\Local\Temp\pip-install-_2xfr3jd\llvmlite_a9473dabcc684c1a9cf12edd5b6d5d0a\ffi\build.py", line 28, in try_cmake
          subprocess.check_call(['cmake', '-G', generator, cmake_dir])
        File "C:\Users\zacha\.conda\envs\CellAssign\lib\subprocess.py", line 364, in check_call
          retcode = call(*popenargs, **kwargs)
        File "C:\Users\zacha\.conda\envs\CellAssign\lib\subprocess.py", line 345, in call
          with Popen(*popenargs, **kwargs) as p:
        File "C:\Users\zacha\.conda\envs\CellAssign\lib\subprocess.py", line 971, in __init__
          self._execute_child(args, executable, preexec_fn, close_fds,
        File "C:\Users\zacha\.conda\envs\CellAssign\lib\subprocess.py", line 1456, in _execute_child
          hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
      FileNotFoundError: [WinError 2] The system cannot find the file specified
      error: command 'C:\\Users\\zacha\\.conda\\envs\\CellAssign\\python.exe' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for llvmlite
  Running setup.py clean for llvmlite
Failed to build numba llvmlite
ERROR: Could not build wheels for numba, llvmlite, which is required to install pyproject.toml-based projects

```

#### Versions:

<!-- Output of scvi.__version__ -->

> VERSION

<!-- Relevant screenshots -->

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -32,6 +32,7 @@ classifiers = [
 ]
 dependencies = [
     "anndata>=0.7.5",
+    "chex<=0.1.8",
     "docrep>=0.3.2",
     "flax",
     "jax>=0.4.4",
@@ -54,7 +55,6 @@ dependencies = [
     "mudata>=0.1.2",
     "sparse>=0.14.0",
     "xarray>=2023.2.0",
-    "chex",
 ]
 
 

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -32,7 +32,6 @@ classifiers = [
 ]
 dependencies = [
     "anndata>=0.7.5",
-    "chex<=0.1.8",
     "docrep>=0.3.2",
     "flax",
     "jax>=0.4.4",
@@ -55,6 +54,7 @@ dependencies = [
     "mudata>=0.1.2",
     "sparse>=0.14.0",
     "xarray>=2023.2.0",
+    "chex",
 ]
 
 

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -32,7 +32,7 @@ classifiers = [
 ]
 dependencies = [
     "anndata>=0.7.5",
-    "chex",
+    "chex<=0.1.8",
     "docrep>=0.3.2",
     "flax",
     "jax>=0.4.4",

